beacon/src/beacon.rs

```diff
@@ -42,24 +42,30 @@ impl Beacon {
                     local_entropy.digest(&mut hasher);
                     hasher.finalize().to_vec()
                 };
-                let mut seed = [0u8; 32];
-                seed.copy_from_slice(&data[0..32]);
-                let mut rng = rand_chacha::ChaCha12Rng::from_seed(seed);
 
-                let frequency = rand_frequency(region_params, &mut rng)?;
-                let payload_size =
-                    rng.gen_range(MIN_BEACON_V0_PAYLOAD_SIZE..=MAX_BEACON_V0_PAYLOAD_SIZE);
-                let datarate = DataRate::Sf7bw125;
+                // Truncate data
+                data.truncate(BEACON_PAYLOAD_SIZE);
+
+                // Selet frequency based on the the first two bytes of the
+                // beacon data
+                let freq_seed = LittleEndian::read_u16(&data) as usize;
+                let frequency =
+                    region_params.params[freq_seed % region_params.params.len()].channel_frequency;
+
+                let spreading = region_params.select_spreading(data.len())?.as_str_name();
+                let bandwidth = region_params.bandwitdh()? / 1000; // in khz
+
+                let datarate = DataRate::from_str(&format!("{spreading}BW{bandwidth}"))
+                    .map_err(|_| Error::no_data_rate())?;
+                let conducted_power = region_params.max_conducted_power()?;
```

Comment:
Are we storing max **conducted power** in the region or are we storing max **effective isotropic radiator power**? I'm worried this is getting mixed again.

Reply:
you just keep coming up with new names :-)

Reply:
Resolving. It is indeed conducted power being returned. The function handles the conversion correctly.